components/ContinentScreen/ContinentScreen.datalayer.jsx

```diff
@@ -1,19 +1,22 @@
-import { useLocalSearchParams, useRouter } from "expo-router";
-
 import useFetchEarth from "../../utils/hooks/useFetchEarth";
 import ContinentScreen from "./ContinentScreen";
 
-function ContinentScreenDatalayer() {
-  const params = useLocalSearchParams();
-  const router = useRouter();
+function ContinentScreenDatalayer(props) {
+
+  const {
+    route,
+    navigation,
+  } = props;
+
+  const code = route.params.code;
```

Comment:
recommend switching to either route?.params?.code or a lodash get(route, 'params.code', defaultValue)
been accidentally burned in javascript by an object unexpectedly not containing what I expect